plugin/plugin.go

```diff
@@ -144,87 +146,110 @@ func (p *Plugin) OnTrafficFromClient(
 	return req, nil
 }
 
-// OnTrafficFromServer is called when a response is received by GatewayD from the server.
-func (p *Plugin) OnTrafficFromServer(
-	ctx context.Context, resp *v1.Struct,
-) (*v1.Struct, error) {
-	OnTrafficFromServerCounter.Inc()
-	resp, err := postgres.HandleServerMessage(resp, p.Logger)
-	if err != nil {
-		p.Logger.Info("Failed to handle server message", "error", err)
-	}
+type UpdateCacheRequest struct {
+	serverResponse *v1.Struct
+}
 
-	rowDescription := cast.ToString(sdkPlugin.GetAttr(resp, "rowDescription", ""))
-	dataRow := cast.ToStringSlice(sdkPlugin.GetAttr(resp, "dataRow", []interface{}{}))
-	errorResponse := cast.ToString(sdkPlugin.GetAttr(resp, "errorResponse", ""))
-	request, ok := sdkPlugin.GetAttr(resp, "request", nil).([]byte)
-	if !ok {
-		request = []byte{}
-	}
-	response, ok := sdkPlugin.GetAttr(resp, "response", nil).([]byte)
-	if !ok {
-		response = []byte{}
-	}
-	server := cast.ToStringMapString(sdkPlugin.GetAttr(resp, "server", ""))
+func (p *Plugin) UpdateCache(ctx context.Context) {
```

Comment:
We can unexport this function, as it should be used internally.

Reply:
This is called from `main` actually. If you think this is not okay, we can create a constructor for `Plugin` and run this function in that constructor I think

Reply:
Can you do that instead? 🙈